x-pack/plugins/fleet/server/services/epm/agent/agent.ts

```diff
@@ -58,8 +57,8 @@ function replaceVariablesInYaml(yamlVariables: { [k: string]: any }, yaml: any)
   return yaml;
 }
 
-const maybeEscapeNumericString = (value: string) => {
-  return value.length && !isNaN(+value) ? `"${value}"` : value;
+const maybeEscapeString = (value: string) => {
+  return safeDump(value);
```

Comment:
@nchaulet This seems to do the right thing, but `safeDump()` is given only one string here, not a complete object. I wonder if this is how it is meant to be used -- I didn't find any documentation for that use case.

Reply:
Just tested the `safeDump` and looks like it had a `\n` at the end not sure it will work for our usecase

Reply:
Have a look at the test case output in https://github.com/elastic/kibana/pull/91418/commits/0409c3ed6538c3b4ed89c7d6fd1423c12488467e , it looks ok to me. I'm happy to change it to a simple `"${value}` as well.

Reply:
I did more test and it's breaking some of our templates where we have loop, yes I think we should implement the escape function ourself here.

Reply:
I've changed this to quote all strings containing YAML special characters as listed in https://yaml.org/spec/1.2/spec.html#id2772075

@nchaulet can you share what you're testing with, and which templates are being broken?